### DISCUSSION TITLE
Image Paste From Clipboard Support for Mac

### DISCUSSION THREAD

**Discussion by MattFlower** [nvim-telekasten/telekasten.nvim#88] [General]
Hello, I've been working on some code locally to improve telekasten.  Specifically, I've implemented:

- a working paste from a mac using only built-in software (osascript)
- better handling of relative paths for image pasting -- you should be able to save the image and file anywhere on your drive, and the relative path should work correctly.
- Improved error handling when you don't have an image on the clipboard, and it can't be converted to a png.

Is this something you'd be interested in?  If so, should I open a PR or do you want to discuss it here in more detail first?

Thanks,
Matt

**Comment by lambtho12** [comment]
I think these are all fantastic ideas, ideally Telekasten should work properly on all platforms.

Regarding point 2:
> better handling of relative paths for image pasting -- you should be able to save the image and file anywhere on your drive, and the relative path should work correctly

I thought I fixed that with #73 . I am not using images that much, so I have only tested it on some limited cases I guess. Are there any issues left?

**Reply by MattFlower** [reply]
I saw a problem with the relpath when you have a markdown file in daily and you have a image_subdir defined.  This was because the relpath was constructed based on the assumption that the file is in home. The png was stored correctly, but you couldn’t actually see it when you went to Markdown Preview.  The relpath was being inserted as “images/pasted_img_…” and was missing the leading dots. 

Here’s the bad news though. I tried using Plenary Path to solve the relpath problem, but its algorithm didn’t seem smart enough to handle the daily/image_subdir case. Instead it was giving the absolute path for the image file as the relpath. I looked at their source code, I don’t think it’s built to make relative paths than require “../“.  I’m fairly handy with algorithms so I was able to write one for this without trouble, but not sure how you’ll feel about taking on another 40 lines of code just to construct a proper relative path. 

I’ve opened up PR #89 so you can get a better idea of what I’m talking about in regards to the additional method.
